fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/codec/LuceneOptimizedBlendedInfixSuggesterWithoutTermVectors.java

```diff
@@ -260,8 +266,31 @@ protected IndexWriterConfig getIndexWriterConfig(Analyzer indexAnalyzer, IndexWr
 
             @Override
             public synchronized void merge(final MergeSource mergeSource, final MergeTrigger trigger) throws IOException {
-                LOGGER.trace("Auto-complete index mergeSource={}", mergeSource);
-                super.merge(mergeSource, trigger);
+                if (state.context.getPropertyStorage().getPropertyValue(LuceneRecordContextProperties.LUCENE_MULTIPLE_MERGE_OPTIMIZATION_ENABLED) && trigger == MergeTrigger.FULL_FLUSH) {
+                    if ((new Random()).nextInt(mergeDirectoryCount) == 1) {
```

Comment:
Should this check for when the random value is equal to 0 instead of 1? I think if you have 1 directory to merge, you might end up never merging

Reply:
Fixed.